proto/Agent.proto

```diff
@@ -215,3 +254,96 @@ message DeleteKeyRequestMessage {
 message DeleteKeyResponseMessage {
   bool successful = 1;
 }
+
+// ==== PeerInfo ==== //
+message PeerInfoRequestMessage {
+  bool current = 1;
+  string public_key = 2;
+}
+message PeerInfoResponseMessage {
+  string public_key = 1;
+  string connected_address = 2;
+  string relay_public_key = 3;
+}
+
+// ==== AddPeer ==== //
+message AddPeerRequestMessage {
+  string public_key = 1;
+  string connected_address = 2;
+  string relay_public_key = 3;
+}
+message AddPeerResponseMessage {
+  bool successful = 1;
+}
+
+// ==== PingPeer ==== //
+message PingPeerRequestMessage {
+  string public_key = 1;
+  int32 timeout = 2;
+}
+message PingPeerResponseMessage {
+  bool successful = 1;
+}
+
+// ==== FindPeer ==== //
+message FindPeerRequestMessage {
+  string public_key = 1;
+  int32 timeout = 2;
+}
+message FindPeerResponseMessage {
+  bool successful = 1;
+}
+
+// ==== FindPeer ==== //
+message FindSocialPeerRequestMessage {
+  string handle = 1;
+  string service = 2;
+  int32 timeout = 3;
+}
+message FindSocialPeerResponseMessage {
+  bool successful = 1;
+}
+
+// ==== ListPeers ==== //
+message ListPeersRequestMessage {
+}
+message ListPeersResponseMessage {
+  repeated string public_keys = 1;
+}
+
+
+// ==== ToggleStealth ==== //
+message ToggleStealthRequestMessage {
+  bool active = 1;
+}
+message ToggleStealthResponseMessage {
+  bool successful = 1;
+}
+
+// ==== UpdatePeerInfo ==== //
+message UpdatePeerInfoRequestMessage {
+  string public_key = 1;
+  bool current_node = 2;
+  string connected_host = 3;
+  int32 connected_port = 4;
+  string relay_public_key = 5;
+}
+message UpdatePeerInfoResponseMessage {
+  bool successful = 1;
+}
+
+// ==== RequestRelay ==== //
+message RequestRelayRequestMessage {
+  string public_key = 1;
+}
+message RequestRelayResponseMessage {
+  bool successful = 1;
+}
+
+// ==== RequestPunch ==== //
+message RequestPunchRequestMessage {
+  string public_key = 1;
+}
+message RequestPunchResponseMessage {
```

Comment:
Since we are doing modelling for the Introcept. I think we can take those ideas and apply it here as well. So if we can create a xstate model that showcases all the communication transitions, that would be good.

Reply:
added a new issue along these lines: https://github.com/MatrixAI/js-polykey/issues/96